blockchain/source/contracts/Market.sol

```diff
@@ -697,41 +537,49 @@ contract Market is Ownable, Pausable {
         return true;
     }
 
-    function RefundRemainingFunds(uint dealID) internal returns (bool){
-        if (deals[dealID].blockedBalance != 0) {
-            token.transfer(deals[dealID].consumerID, deals[dealID].blockedBalance);
-            deals[dealID].blockedBalance = 0;
+    function RefundRemainingFunds(uint dealID) internal returns (bool) {
+        address consumerID = dl.GetDealConsumerID(dealID);
+
+        uint blockedBalance = dl.GetDealBlockedBalance(dealID);
+
+        if (blockedBalance != 0) {
+            token.transfer(consumerID, blockedBalance);
+            dl.SetDealBlockedBalance(dealID, 0);
         }
         return true;
     }
 
-    function IsSpot(uint dealID) internal view returns (bool){
-        return deals[dealID].duration == 0;
-    }
+
 
     function CalculatePayment(uint _price, uint _period) internal view returns (uint) {
         uint rate = oracle.getCurrentPrice();
         return rate.mul(_price).mul(_period).div(1e18);
     }
 
     function AddToBlacklist(uint dealID, BlacklistPerson role) internal {
+        (, address supplierID, address consumerID, address masterID, , , ) = dl.GetDealInfo(dealID);
```

Comment:
you have getters for this vars;

Reply:
yes, but I also have one getter for all of them